packages/components/src/text-control/stories/index.tsx

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+import type { ComponentMeta, ComponentStory } from '@storybook/react';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import TextControl from '..';
+
+const meta: ComponentMeta< typeof TextControl > = {
+	component: TextControl,
+	title: 'Components/TextControl',
+	argTypes: {
+		onChange: {
+			action: 'onChange',
+			control: { type: null },
+		},
+		value: {
+			control: { type: null },
+		},
+	},
+	parameters: {
+		controls: {
+			expanded: true,
+		},
+		docs: { source: { state: 'open' } },
+	},
+};
+export default meta;
+
+const DefaultTemplate: ComponentStory< typeof TextControl > = ( {
+	...args
+} ) => {
+	const [ value, setValue ] = useState( '' );
+
+	return (
+		<TextControl
+			{ ...args }
+			value={ value }
+			onChange={ ( v ) => {
+				setValue( v );
+			} }
```

Comment:
Apparently, the Action thing (enabled on [this line](https://github.com/WordPress/gutenberg/pull/40633/files#diff-f162be886476a0f6aaa0dc9958d453e89d1e56ce78a01b1918873dc958d51edeR21)) only works by passing through the `onChange` prop.

```diff
const DefaultTemplate: ComponentStory< typeof TextControl > = ( {
+	onChange,
	...args
} ) => {
	const [ value, setValue ] = useState( '' );

	return (
		<TextControl
			{ ...args }
			value={ value }
			onChange={ ( v ) => {
				setValue( v );
+				onChange( v );
```

https://user-images.githubusercontent.com/555336/165803385-b897a8e2-3ad4-44a8-95f6-091672672c9e.mp4

Reply:
Thanks. Added in [e30a8dd](https://github.com/WordPress/gutenberg/pull/40633/commits/e30a8ddf2e7abf6548bf3b71d0d4df7ddf66f280)